backend/molgenis-emx2-semantics/src/main/java/org/molgenis/emx2/beaconv2/endpoints/individuals/ejp_rd_vp/EJP_VP_IndividualsQuery.java

```diff
@@ -0,0 +1,174 @@
+package org.molgenis.emx2.beaconv2.endpoints.individuals.ejp_rd_vp;
+
+import static org.molgenis.emx2.beaconv2.common.QueryHelper.finalizeFilter;
+import static org.molgenis.emx2.beaconv2.common.QueryHelper.findColumnPath;
+import static org.molgenis.emx2.beaconv2.endpoints.individuals.QueryIndividuals.queryIndividuals;
+import static org.molgenis.emx2.json.JsonUtil.getWriter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import org.molgenis.emx2.Table;
+import org.molgenis.emx2.beaconv2.common.ColumnPath;
+import org.molgenis.emx2.beaconv2.endpoints.individuals.IndividualsResultSets;
+import org.molgenis.emx2.beaconv2.requests.BeaconRequestBody;
+import org.molgenis.emx2.beaconv2.requests.Filter;
+import org.molgenis.emx2.beaconv2.responses.BeaconCountResponse;
+import spark.Request;
+import spark.Response;
+
+/** Implements https://github.com/ejp-rd-vp/vp-api-specs */
+public class EJP_VP_IndividualsQuery {
+
+  private Request request;
+  private Response response;
+  private List<Table> tables;
+
+  public EJP_VP_IndividualsQuery(Request request, Response response, List<Table> tables) {
+    this.request = request;
+    this.response = response;
+    this.tables = tables;
+  }
+
+  public String getPostResponse() throws JsonProcessingException {
+
+    BeaconRequestBody beaconRequestBody =
+        new ObjectMapper().readValue(request.body(), BeaconRequestBody.class);
+
+    List<String> filters = new ArrayList<>();
+    for (Filter filter : beaconRequestBody.getQuery().getFilters()) {
+
+      // type is the ontology tag of the column we want to query on
+      // either full or partial: "http://purl.obolibrary.org/obo/NCIT_C48697" or "NCIT_C48697"
+      String type = filter.getType();
+
+      // id is the specific thing to match individuals on, e.g. a particular disease, age of onset,
+      // etc.
+      String id = filter.getId();
+
+      // operator (=, >, !, etc) actually not used in this context
+      String operator = filter.getOperator();
+
+      /**
+       * All of the 'age' related queries. NCIT_C25150 = Age this year, i.e. 'age', EFO_0004847 =
+       * Age at disease manifestation, i.e. 'age of onset', NCIT_C156420 = Age at diagnosis
+       */
+      if (type.endsWith("NCIT_C25150")
+          || type.equals("EFO_0004847")
+          || type.equals("NCIT_C156420")) {
+        int age = Integer.parseInt(id);
+
+        if (type.endsWith("NCIT_C25150")) {
+          // Age (Age this year)
+          String ageFilter =
+              "{ _or: [{age__age__iso8601duration: {like: \"P"
```

Comment:
Again it, would be nice if strings like 'age__age__iso8601duration' are constants, doing so would allow you to remove quite some comments from this file.

Reply:
done!